wordpress.org/public_html/wp-content/plugins/theme-directory/class-wporg-themes-auto-review.php

```diff
@@ -0,0 +1,173 @@
+				array( 'status' => \WP_Http::UNAUTHORIZED )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns whether the ticket number is for the correct theme.
+	 * 
+	 * @param string $keywords A string that should include the theme slug.
+	 * @param string $theme_slug The theme slug.
+	 * @return bool
+	 */
+	public function ticket_is_for_theme( $keywords, $theme_slug ) {
+		return ! empty( $keywords ) && strpos( $keywords , $theme_slug );
+	}
+
+	/**
+	 * Returns a cleaned up version of the content.
+	 * 
+	 * @param string $content GitHub specific language to format.
+	 * @return string
+	 */
+	public function get_formatted_content( $content ) {
+		$padding = PHP_EOL . PHP_EOL;
+
+		// Add some space around the titles
+		$content = str_replace( "]:", "]:" . $padding, $content );
+		$content = str_replace( "[", $padding . "[", $content );
+
+		// Remove empty spaces
+		$content = trim( $content );
+
+		$disclaimer = sprintf( __( "''The following has been generated by https://github.com/WordPress/theme-review-action, this is for informational purposes only.''" . PHP_EOL, 'wporg-themes' ) );
+
+		$content = "
+{$disclaimer}
+{$content}
+";
+		return $content;
+	}
+
+	/**
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function update_item( $request ) {
+		if ( ! class_exists( 'Trac' ) ) {
+			require_once ABSPATH . WPINC . '/class-IXR.php';
+			require_once ABSPATH . WPINC . '/class-wp-http-ixr-client.php';
+			require_once __DIR__ . '/lib/class-trac.php';
+		}
+
+		$trac_instance = new \Trac( 'themetracbot', THEME_TRACBOT_PASSWORD, 'https://themes.trac.wordpress.org/login/xmlrpc' );
+		$theme_slug = $request[ 'theme_slug' ];
+		$ticket_id = $request[ 'ticket_id' ];
+
+		// Check for ticket
+		$ticket = $trac_instance->ticket_get( $ticket_id );
+
+		if( ! $ticket ){
+			return new \WP_Error(
+				'rest_invalid_ticket',
+				__( 'Unable to locate ticket.', 'wporg-themes' ),
+				array( 'status' => \WP_Http::NOT_FOUND )
+			);
+		}
+
+		// Check to make sure ticket matches theme
+		if( ! $this->ticket_is_for_theme( $ticket['keywords'], $theme_slug ) ) {
+			return new \WP_Error(
+				'rest_invalid_ticket_for_theme',
+				__( 'Ticket does not match theme.', 'wporg-themes' ),
+				array( 'status' => \WP_Http::BAD_REQUEST )
+			);
+		}
+
+		$body_params =  $request->get_body_params();
+		$content = $this->get_formatted_content( $body_params[ 'content' ] );
+
+		$updated_ticket = $trac_instance->ticket_update( $ticket_id, $content, [], false );
+
+		if( ! $updated_ticket ) {
+			return new \WP_Error(
+				'rest_error_updating_ticket',
+				__( 'We ran into an error updating the ticket.', 'wporg-themes' ),
+				array( 'status' => \WP_Http::INTERNAL_SERVER_ERROR )
+			);
+		}
+
+		return new WP_REST_Response( __( 'Success.', 'wporg-themes' ), \WP_Http::OK );
```

Comment:
Simply `return true;` is probably enough here, there's probably no need to return a human-readable success message?

Reply:
I think it's good to include the `200` status code, since API clients will often explicitly look for that.

Reply:
I was under the impression that passing `\WP_Http::OK` as the second param will set the status code to `200`?

Reply:
Ah, in relation to @dd32's comment... I do want to return a proper `200` status, as @coreymckrill mentions. In case the client throws an exception for not receiving a 200.

Reply:
Oh yep, `\WP_Http::OK` is fine. I just meant I think it should include a status code.